sdk/core/core-http/src/serviceClient.ts

```diff
@@ -531,6 +534,10 @@ export function serializeRequestBody(
               rootName: xmlName || serializedName
             });
           }
+        } else if (typeName === "String" && operationSpec.contentType?.match("text/plain")) {
+          // the String serializer has validated that request body is a string
+          // so just send the string.
+          return;
```

Comment:
while you're in this code, could you go and drop "text/plain" from here:

https://github.com/Azure/azure-sdk-for-js/blob/4555a454b5effde0d7e52b39e6b4996861c4d8d1/sdk/core/core-http/src/policies/deserializationPolicy.ts#L61

This is something that individual clients (like Identity) should customize as they need to.

Reply:
I would rather not touch any logic about deserialization in this PR if that's ok?

Reply:
sure, but "support text/plain endpoints" makes it sound like you wanted to handle text/plain results too 😉 

Language lawyer nit: `[core-http] Properly format text/plain request bodies`

Reply:
Well that makes me nervous that we treat text/plain as a default JSON content type. I wonder if the same assumption is made for requests (@bterlson any chance you tested autorest.typescript v5 to see how it handles text/plain on requests?)

Reply:
I updated the title of this PR.

Reply:
> Well that makes me nervous that we treat text/plain as a default JSON content type. 

I seem to recall that this was a fix for a specific service that had this. I agree it is worrying...

> @bterlson any chance you tested autorest.typescript v5 to see how it handles text/plain on requests?)

I did not, but I can't imagine it worked without a change like above?

Reply:
The change I wanted has its own PR now: https://github.com/Azure/azure-sdk-for-js/pull/7965